### PR TITLE
Running forge with latest solc

### PR DESCRIPTION
**Pull request by aroralanuk**
### Description

- Removed contracts-bedrock contracts dependency from eth-optimism (which was forcing us to use 0.8.15)
- Added local minimal interfaces for the CrossDomainMessenger contract

### Drive-by changes

None

### Related issues

None

### Backward compatibility

Yes

### Testing

Unit tests

### CODE REVIEW

File: solidity/foundry.toml [hunk 1]
```diff
@@ -3,9 +3,8 @@ src = 'contracts'
 out = 'out'
 libs = ['node_modules', 'lib']
 test = 'test'
-cache_path  = 'forge-cache'
+cache_path = 'forge-cache'
 allow_paths = ["../node_modules"]
-solc = '0.8.15'
```
Comment (yorhodes) [inline solidity/foundry.toml:8]: we should specify still no?

File: solidity/test/isms/OPStackIsm.t.sol [hunk 2]
```diff
@@ -154,17 +149,15 @@ contract OPStackIsmTest is Test {
             (messageId)
         );
 
-        uint40 nonce = ICanonicalTransactionChain(L1_CANNONICAL_CHAIN)
```
Comment (yorhodes) [inline solidity/test/isms/OPStackIsm.t.sol:157]: aren't we fork testing?

File: solidity/test/isms/OPStackIsm.t.sol [hunk 3]
```diff
@@ -232,15 +225,12 @@ contract OPStackIsmTest is Test {
             (messageId)
         );
 
-        (uint240 nonce, uint16 verison) = Encoding.decodeVersionedNonce(
+        (uint240 nonce, uint16 verison) = decodeVersionedNonce(
```
Comment (yorhodes) [inline solidity/test/isms/OPStackIsm.t.sol:228]: nit: spelling here and everywhere

File: solidity/test/isms/OPStackIsm.t.sol [hunk 4]
```diff
@@ -232,15 +225,12 @@ contract OPStackIsmTest is Test {
...
             l2Messenger.messageNonce()
         );
-        uint256 versionedNonce = Encoding.encodeVersionedNonce(
-            nonce + 1,
-            verison
-        );
+        uint256 versionedNonce = encodeVersionedNonce(nonce + 1, verison);
 
-        bytes32 versionedHash = Hashing.hashCrossDomainMessageV1(
+        bytes32 versionedHash = hashCrossDomainMessageV1(
```
Comment (yorhodes) [inline solidity/test/isms/OPStackIsm.t.sol:233]: imo this is a nice opportunity to do the prank and remove the forking but not gonna force it

File: solidity/test/isms/OPStackIsm.t.sol [hunk 5]
```diff
@@ -465,4 +443,102 @@ contract OPStackIsmTest is Test {
                 testMessage
             );
     }
+
+    /// @dev from eth-optimism/contracts-bedrock/contracts/libraries/Hashing.sol
+    /// @notice Hashes a cross domain message based on the V1 (current) encoding.
```
Comment (yorhodes) [inline solidity/test/isms/OPStackIsm.t.sol:448]: would be nice not to vendor this ;)

**Review by yorhodes — APPROVED** [review]

File: solidity/foundry.toml (hunk 1 above)
Comment (aroralanuk) [reply solidity/foundry.toml:8]: No, we should let it infer it

File: solidity/test/isms/OPStackIsm.t.sol (hunk 2 above)
Comment (aroralanuk) [reply solidity/test/isms/OPStackIsm.t.sol:157]: we're not testing the nonce in the event

**Pull request merged**